### PR TITLE
Added GitHub Actions to automate packaging

### PR DESCRIPTION
**Pull request by penev92**
Same as https://github.com/OpenRA/Eluant/pull/5 and https://github.com/OpenRA/OpenAL-CS/pull/4:

Adding 3 workflows to automate compiling of native dependencies, compiling of the C# wrapper library and packaging it all into a NuGet package and publishing that.

The native binary compilation steps are all taken from @pchote's https://github.com/pchote/native-dependencies

P.S.: For results of a successful run see https://github.com/penev92/SDL2-CS/actions/runs/4780366537 and https://www.nuget.org/packages/OpenRA-SDL2-CS/1.0.38-a9

P.P.S.: I added a few improvements over the OpenAL-CS PR which I plan on porting back later to https://github.com/OpenRA/OpenAL-CS and https://github.com/OpenRA/Eluant.

### CODE REVIEW

**Review by pchote — APPROVED** [review]
macos-arm64 and linux-arm64 both seem to work using the -a9 nuget.

File: .github/workflows/build-natives.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                lipo -thin arm64 lib/libSDL2-2.0.0.dylib -output ../../artifacts/arm64/SDL2.dylib
+
+            - name: Upload Artifacts
+              uses: actions/upload-artifact@v3
+              with:
+                name: Natives-MacOS
+                path: ./artifacts
+
+    # Note: Running inside a centos container because we want to compile using a version of glibc that is as old as reasonably possible
+    # to ensure backwards compatibility of the combiled binaries.
```
Comment (abcdefg30) [inline .github/workflows/build-natives.yml]:
```suggestion
    # to ensure backwards compatibility of the compiled binaries.
```

**Review by abcdefg30 — APPROVED** [review]

**Comment by Mailaender** [comment]
I suggest you download the `libEGL*.dll` from Google inside the workflow so we know where they are coming from. Then you also don't have binaries in git which is bad practice.

**Review by Mailaender — CHANGES_REQUESTED** [review]
I suggest you download the `libEGL*.dll` from Google inside the workflow so we know where they are coming from. Then you also don't have binaries in git which is bad practice.

**Comment by penev92** [comment]
Like I said I would prefer that as well, but do we have a way to do that?

**Comment by Mailaender** [comment]
Like I said yesterday, either extract from [Chromium ZIPs](https://storage.googleapis.com/chromium-browser-snapshots/Win_x64/1135051/chrome-win.zip) during the Workflow or [build yourself](https://chromium.googlesource.com/angle/angle/+/master/doc/DevSetup.md).

**Comment by penev92** [comment]
The links with the zips that you gave me on Discord had Chromium releases up until 2018!
As for compiling it ourselves - I did read the "instructions" and they made sure it was as convoluted as possible. It's out of my depth and I don't want to spend weeks struggling with it.

**Comment by Mailaender** [comment]
The link I gave you above is from https://chromium.woolyss.com/download/ which is dated to yesterday.

**Comment by penev92** [comment]
Thanks!
Here is the result of the extracting - you can see execution time, the zipped artifact and the used workflow file: https://github.com/penev92/SDL2-CS/actions/runs/4819116799
I will update the PR when I have some time.

**Comment by penev92** [comment]
Fixed typos, removed ANGLE libraries for Windows and added downloading of Chromium to extract them.

**Comment by penev92** [comment]
Here is the run for `1.0.38-a10`: https://github.com/penev92/SDL2-CS/actions/runs/4860223694
Link to NuGet package: https://www.nuget.org/packages/OpenRA-SDL2-CS/1.0.38-a10

**Review by Mailaender — APPROVED** [review]

**Pull request merged**